interfaces/builtin/avahi_control_test.go

```diff
@@ -0,0 +1,181 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type AvahiControlInterfaceSuite struct {
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&AvahiControlInterfaceSuite{
+	iface: builtin.MustInterface("avahi-control"),
+})
+
+const avahiControlMockPlugSnapInfoYaml = `name: other
+version: 1.0
+apps:
+ app2:
+  command: foo
+  plugs: [avahi-control]
+`
+const avahiControlMockSlotSnapInfoYaml = `name: avahi
+version: 1.0
+apps:
+ app1:
+  command: foo
+  slots: [avahi-control]
+`
+
+func (s *AvahiControlInterfaceSuite) SetUpTest(c *C) {
+	slotSnap := snaptest.MockInfo(c, avahiControlMockSlotSnapInfoYaml, nil)
+	s.slot = &interfaces.Slot{SlotInfo: slotSnap.Slots["avahi-control"]}
+	plugSnap := snaptest.MockInfo(c, avahiControlMockPlugSnapInfoYaml, nil)
+	s.plug = &interfaces.Plug{PlugInfo: plugSnap.Plugs["avahi-control"]}
+}
+
+func (s *AvahiControlInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "avahi-control")
+}
+
+// The label glob when all apps are bound to the avahi slot
+func (s *AvahiControlInterfaceSuite) TestConnectedPlugSnippetUsesSlotLabelAll(c *C) {
```

Comment:
Can you please use `snaptest.MockInfo` and extract those from the resulting `snap.Info` instead?